src/AaveV3StrategicAssets_20230606/VeTokenManager.sol

```diff
@@ -0,0 +1,219 @@
+      minPerc,
+      maxPerc,
+      useAdvicePrice
+    );
+    emit SellBoost(pricePerVote, maxDurationWeeks, expiryTime, minPerc, maxPerc, useAdvicePrice);
+  }
+
+  /// @notice Removes existing boost offer
+  function removeBoostOffer() external onlyOwnerOrGuardian {
+    IERC20(IWardenBoost(WARDEN_VE_BAL).delegationBoost()).approve(WARDEN_VE_BAL, 0);
+    IWardenBoost(WARDEN_VE_BAL).quit();
+    emit RemoveBoostOffer();
+  }
+
+  /// @notice Claim fee token rewards accrued by selling veBoost
+  function claimBoostRewards() external onlyOwnerOrGuardian {
+    IWardenBoost(WARDEN_VE_BAL).claim();
+    emit ClaimBoostRewards();
+  }
+
+  /// @notice sets the snapshot space ID
+  /// @param _spaceId The string representation of the spaceId
+  function setSpaceIdVEBAL(bytes32 _spaceId) external onlyOwnerOrGuardian {
+    DELEGATE_REGISTRY.clearDelegate(spaceIdBalancer);
+    spaceIdBalancer = _spaceId;
+    emit SetSpaceId(_spaceId);
+    _delegate(delegate);
+  }
+
+  /// @notice sets the snapshot delegate
+  /// @param newDelegate Address of the user to delegate to
+  function setDelegateVEBAL(address newDelegate) external onlyOwnerOrGuardian {
+    _delegate(newDelegate);
+  }
+
+  /// @notice clears the delegate from snapshot
+  function clearDelegateVEBAL() external onlyOwnerOrGuardian {
+    address oldDelegate = delegate;
+    DELEGATE_REGISTRY.clearDelegate(spaceIdBalancer);
+
+    emit DelegateUpdate(oldDelegate, address(0));
+  }
+
+  /// @notice Increases the lock duration for underlying token
+  /// @param newLockDuration New lock duration for underlying-locking
+  function setLockDurationVEBAL(uint256 newLockDuration) external onlyOwnerOrGuardian {
+    lockDurationVEBAL = newLockDuration;
+    emit SetLockDuration(newLockDuration);
+  }
+
+  /// @notice Locks underlying token into veToken until lockDuration
+  /// @notice Increases amount or lock duration if lock already exists
+  function lockVEBAL() external onlyOwnerOrGuardian {
+    uint256 tokenBalance = IERC20(B_80BAL_20WETH).balanceOf(address(this));
+    uint256 locked = IVeToken(VE_BAL).locked(address(this));
+    uint256 lockHorizon = ((block.timestamp + lockDurationVEBAL) / WEEK) * WEEK;
+
+    if (tokenBalance != 0 && locked == 0) {
+      // First lock
+      IERC20(B_80BAL_20WETH).approve(VE_BAL, tokenBalance);
+      IVeToken(VE_BAL).create_lock(tokenBalance, lockHorizon);
+    } else if (tokenBalance != 0 && locked != 0) {
+      // Increase amount of tokens locked & refresh duration to lockDuration
+      IERC20(B_80BAL_20WETH).approve(VE_BAL, tokenBalance);
+      IVeToken(VE_BAL).increase_amount(tokenBalance);
+      if (IVeToken(VE_BAL).locked__end(address(this)) != lockHorizon) {
+        IVeToken(VE_BAL).increase_unlock_time(lockHorizon);
+      }
+    } else if (tokenBalance == 0 && locked != 0) {
+      // No additional tokens to lock, just refresh duration to lockDuration
+      IVeToken(VE_BAL).increase_unlock_time(lockHorizon);
+    } else {
```

Comment:
Shouldn't this revert? Why just do nothing?

Reply:
hmm, well there's nothing to do. reverting seemed weird but can add a revert that says something like nothing to lock or relock.

Reply:
Fair enough, I think it's fine as is